### DISCUSSION TITLE
About migration from v2.31 to v3.3

### DISCUSSION THREAD

**Discussion by andyyumiao** [Netflix/conductor#2546] [Q&A] [answered]
The version we are currently using is relatively old version 2.31. As I know, the latest version 3.3 has made many changes, such as the deletion of the primary key id in the database and the adjustment of task definitions (DECISION is adjusted to Switch, and LAMBDA is adjusted to INLINE
), so my question is how to upgrade smoothly? for example:
1. Our production environment has a large number of workflow templates, which use DECISION and LAMBDA tasks. Is the new version compatible? If not, does it mean that I need to update all these templates?
2. Are there any other things I need to pay attention to?

**Comment by buzz08** [comment, marked as answer]
DECISION and LAMBDA are deprecated, but still available in the latest version. We have tested this and it is working. But ideally you should update to the newer versions.

**Comment by buzz08** [comment]
@aravindanr - Was there a migration guide published from 2.x to 3.x?

**Comment by aravindanr** [comment]
[SWITCH](https://netflix.github.io/conductor/configuration/systask/#switch) and [INLINE](https://netflix.github.io/conductor/configuration/systask/#inline-task) were introduced as part of [3.2.0](https://github.com/Netflix/conductor/releases/tag/v3.2.0). There is not a migration guide per se, as its self-explanatory from the documentation of those tasks.
